### PR TITLE
Add switch for modular implicits variant with let operator syntax

### PR DESCRIPTION
**Pull request by pxeger**
It works, the tests pass, and it bootstraps itself.

https://github.com/modular-implicits/backports

### CODE REVIEW

**Comment by pxeger** [comment]
Actually, the two features don't interact properly together yet.

```ocaml
val (let/) : {M : Imp.Control.Monad} -> 'a M.t -> ('a -> 'b M.t) -> 'b M.t

let/ x = Some 2 in return x
```

```
Error: The operator let/ has type
         {M : Imp.Control.Monad} -> 'a M.t -> ('a -> 'b M.t) -> 'b M.t
       but it was expected to have type 'c -> ('d -> 'e) -> 'f
```

**Review by yallop — COMMENTED** [review]
The "version number" part (`let_syntax`) in the `base-implicits.let_syntax` package suggests that the package is related to the let syntax, but the package metadata itself seems more general.

(It's likely to be tricky to maintain the invariant across all the implicits-compilers, though, since the other packages are part of the main opam repository.)

**Comment by pxeger** [comment]
Should we just merge this, knowing that it's merely an MVP implementation?

File: packages/base-implicits/base-implicits.let_syntax/opam [hunk 1]
```diff
@@ -0,0 +1,17 @@
+opam-version: "2.0"
+maintainer: "yallop@gmail.com"
+authors: ["Jeremy Yallop"]
```
Comment (yallop) [inline packages/base-implicits/base-implicits.let_syntax/opam:3]:
```suggestion
authors: ["Patrick Reader"]
```

**Review by yallop — CHANGES_REQUESTED** [review]
Could you also please add [CI configuration](https://github.com/modular-implicits/modular-implicits-opam/tree/master/.github/workflows) (including a badge in the README) for the new switch?

**Comment by pxeger** [comment]
Before merging I will need to change https://github.com/modular-implicits/modular-implicits-opam/pull/7/files#diff-c4d4a4472caf54b259ef50bf892777a16a577791095ed7dc10a740fab65e2060R28 to `#master`, to break the catch-22, as discussed

**Comment by pxeger** [comment]
Looks like `[skip CI]` didn't work. I will need admin permissions to merge this because of the catch-22. Or you can just do it.

**Pull request merged**